src/SourceSaver.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+
+namespace Armandsar\LaravelTranslationio;
+
+use Armandsar\LaravelTranslationio\PrettyVarExport;
+use Illuminate\Contracts\Foundation\Application;
+use Illuminate\Filesystem\Filesystem;
+
+class SourceSaver
+{
+    /**
+     * @var Application
+     */
+    private $application;
+    /**
+     * @var Filesystem
+     */
+    private $filesystem;
+    /**
+     * @var PrettyVarExport
+     */
+    private $prettyVarExport;
+
+    public function __construct(
+        Application $application,
+        FileSystem $fileSystem,
+        PrettyVarExport $prettyVarExport
+    )
+    {
+        $this->application = $application;
+        $this->filesystem = $fileSystem;
+        $this->prettyVarExport = $prettyVarExport;
+    }
+
+    public function call($sourceEdit, $sourceLocale)
+    {
+        $dir = $this->localePath($sourceLocale);
+
+        $this->makeDirectoryIfNotExisting($dir);
+
+        $group = $this->group($sourceEdit['key']);
+        $groupFile = $dir . DIRECTORY_SEPARATOR . $group . '.php';
+
+        if ($this->filesystem->exists($groupFile)) {
+            $translations = $this->filesystem->getRequire($groupFile);
+
+            $this->replaceValueInTranslations($translations, $sourceEdit);
+
+            $fileContent = <<<'EOT'
+<?php
+return {{translations}};
+EOT;
+
+            $prettyTranslationsExport = $this->prettyVarExport->call($translations, ['array-align' => true]);
+            $fileContent  = str_replace('{{translations}}', $prettyTranslationsExport, $fileContent);
+
+            $this->filesystem->put($groupFile, $fileContent);
+        }
+    }
+
+
+    private function localePath($locale)
+    {
+        return $this->path() . DIRECTORY_SEPARATOR . $locale;
+    }
+
+    private function path()
+    {
+        return $this->application['path.lang'];
+    }
+
+    private function makeDirectoryIfNotExisting($directory)
+    {
+        if ( ! $this->filesystem->exists($directory)) {
+            $this->filesystem->makeDirectory($directory);
+        }
+    }
+
+    private function group($key)
+    {
+        return explode('.', $key)[0];
+    }
+
+    private function keys($key)
+    {
+        $keyParts = explode('.', $key);
+        array_shift($keyParts); // remove group part
+        return $keyParts;
+    }
+
+    private function replaceValueInTranslations(&$translations, $sourceEdit)
```

Comment:
I dont really like the use of "reference" here, but as long as this works no big deal, we can return back to this and refactor when we have tests covering this part